### PR TITLE
Fixing issue filter

### LINKED ISSUE #24846: Issue filter list looks not good on phone
### Description

image

https://try.gitea.io/yp05327/longnamelongnamelongnamelongnamelongnamelongnamelongnamelongname/issues

### Gitea Version

latest

### Can you reproduce the bug on the Gitea demo site?

Yes

### Log Gist

_No response_

### Screenshots

_No response_

### Git Version

_No response_

### Operating System

_No response_

### How are you running Gitea?

try.gitea.io

### Database

None

### PR DESCRIPTION
**Pull request by puni9869**
In current version of Gitea we have problems with mobile viewport.
Current Mobile screen:
image

## After Modification Mobile Screen [Main Modification]:
image

## Dropdowns:
image

## Ultra Large Devices: 
image

## Large Devices:
image

## Devices less than 1024 and grater than mobile screen:
image

Fixes https://github.com/go-gitea/gitea/issues/24846

### CODE REVIEW

**Comment by puni9869** [comment]
This is a very complex style issue to fix. But I have tried my best to cover all the screen sizes, specially mobile view and viewport < 1024px .  Obviosly there are some tricks and hacks I have used.

I know there may be some changes which not looks great in terms of coding style. I am open for all the feedbacks and suggestions to make the screen great.

## Please take the pull of this branch and run at the local while reviewing. Code may looks great but keeping ui  screen on side by side will give extra points to consider why this style used.

### PS: I am not super great with css. But if any one want to help me in fixing the edge cases I will be greatful.

**Comment by silverwind** [comment]
I think we could also just make those wrap instead. Horizontal scrollbars are suboptimal UX on mobile, and I would also like to eventually get rid of the one on the repo tab bar.

**Comment by puni9869** [comment]
> This is a very complex style issue to fix. But I have tried my best to cover all the screen sizes, specially mobile view and viewport < 1024px . Obviosly there are some tricks and hacks I have used.
> 
> I know there may be some changes which not looks great in terms of coding style. I am open for all the feedbacks and suggestions to make the screen great.
> 
> ## Please take the pull of this branch and run at the local while reviewing. Code may looks great but keeping ui screen on side by side will give extra points to consider why this style used.

Sure thing, 
so you are referring to a vertical collapsed nav bar for repo tabs

**Comment by silverwind** [comment]
Yes, for this I want to make a JS overflow solution into a "..." button similar to GitHub, but it's not for this PR:

image

For this, we can have a horizontal flexbox with `gt-fw` to let them wrap.

image

**Comment by puni9869** [comment]
> Yes, for this I want to make a JS overflow solution into a "..." button similar to GitHub, but it's not for this PR:
> 
> image
> For this, we can have a horizontal flexbox with `gt-fw` to let them wrap.
> 
> image

Got it,
image

we can target in next followup pr.

File: web_src/css/repo/issue-list.css [hunk 1]
```diff
@@ -1,15 +1,61 @@
...
+  #issue-filters > .column.six.wide.column {
+    max-width: fit-content;
+  }
+
+  #issue-filters > .issue-list-toolbar-right > .stackable.menu {
+    margin-left: calc(33% - 7vw);
+  }
+}
+
+@media only screen and (min-width: 1266px) {
```
Comment (silverwind) [inline web_src/css/repo/issue-list.css]: This 1266px does not match any of our other media queries, I think the right one is 1200px.

File: web_src/css/repo/issue-list.css [hunk 2]
```diff
@@ -1,15 +1,61 @@
...
+  #issue-filters > .issue-list-toolbar-right > .stackable.menu {
+    margin-left: 14vh;
+  }
```
Comment (silverwind) [inline web_src/css/repo/issue-list.css:24]: Surely there must be better values like pixels than setting a margin to viewport height, no?

File: web_src/css/repo/issue-list.css [hunk 3]
```diff
@@ -1,15 +1,61 @@
...
 
-.issue-list-toolbar-left {
-  display: flex;
+@media only screen and (max-width: 767.98px) {
+  .ui.container > .ui.stackable.grid > .column.issue-list-toolbar-right > .stackable.menu {
+    overflow-y: hidden;
+    overflow-x: scroll;
+    flex-direction: row !important;
+    flex-wrap: nowrap !important;
+  }
```
Comment (silverwind) [inline web_src/css/repo/issue-list.css:41]: Too long selectors, give it a unique class instead.

File: web_src/css/repo/issue-list.css [hunk 4]
```diff
@@ -1,15 +1,61 @@
...
-  align-items: flex-start;
-  gap: 1rem;
-  margin-top: 1rem;
+@media only screen and (min-width: 1024px) and (max-width: 1266px) {
+  #issue-filters > .column.six.wide.column {
+    max-width: fit-content;
+  }
+
+  #issue-filters > .issue-list-toolbar-right > .stackable.menu {
+    margin-left: calc(33% - 7vw);
```
Comment (silverwind) [inline web_src/css/repo/issue-list.css]: Same here, no viewport based margin please.

File: templates/repo/issue/list.tmpl [hunk 5]
```diff
@@ -81,14 +81,15 @@
 				{{end}}
 			{{end}}
 		</div>
-		<div id="issue-filters" class="issue-list-toolbar">
-			<div class="issue-list-toolbar-left">
+		<div class="ui divider"></div>
```
Comment (silverwind) [inline templates/repo/issue/list.tmpl]: IIRC, I recently removed these dividers to save vertical space. Would prefer to not have them re-introduced, unless there is a very good reason.

Comment (puni9869) [reply templates/repo/issue/list.tmpl]: Done, removed

File: web_src/css/repo/issue-list.css (hunk 4 above)
Comment (puni9869) [reply web_src/css/repo/issue-list.css]: Changed

File: web_src/css/repo/issue-list.css (hunk 1 above)
Comment (puni9869) [reply web_src/css/repo/issue-list.css]: Changed to 1200px.

File: web_src/css/repo/issue-list.css (hunk 2 above)
Comment (puni9869) [reply web_src/css/repo/issue-list.css:24]: Changed

**Comment by puni9869** [comment]
@wxiaoguang  
Requesting for your feedback.

**Comment by wxiaoguang** [comment]
Thank you for inviting me for reviewing, and thank you for your contribution 🙏  . I am pretty busy on this one (`25330`) and its related problems, I haven't got enough time for other PRs at the moment.

Overall I think I am neutral to these changes, while I am not sure whether different languages would also affect the layout. I might take a look later but I can't promise because I am going be in travel in the following weeks.

File: templates/repo/issue/list.tmpl [hunk 6]
```diff
@@ -81,15 +81,15 @@
 				{{end}}
 			{{end}}
 		</div>
-		<div id="issue-filters" class="issue-list-toolbar">
-			<div class="issue-list-toolbar-left">
+		<div id="issue-filters" class="ui stackable grid">
+			<div class="six wide column issue-list-toolbar-left">
```
Comment (wxiaoguang) [inline templates/repo/issue/list.tmpl]: I am not sure whether it needs "stackable grid", it looks like almost every style is defined by our code now, but not by Fomantic's mechanism?

File: templates/repo/issue/list.tmpl [hunk 7]
```diff
@@ -81,15 +81,15 @@
...
 				{{if and ($.CanWriteIssuesOrPulls) (gt (len .Issues) 0)}}
 					<input type="checkbox" autocomplete="off" class="issue-checkbox-all gt-mr-4" title="{{.locale.Tr "repo.issues.action_check_all"}}">
 				{{end}}
 				{{template "repo/issue/openclose" .}}
 			</div>
-			<div class="issue-list-toolbar-right">
-				<div class="ui secondary filter stackable menu labels">
+			<div class="ten wide right aligned column issue-list-toolbar-right">
+				<div class="ui secondary filter stackable menu labels issues-menu">
```
Comment (wxiaoguang) [inline templates/repo/issue/list.tmpl]: Does it needs this new "issues-menu"? It looks like ".issue-list-toolbar-right > .filter.menu" is already for it?

File: web_src/css/repo/issue-list.css [hunk 8]
```diff
@@ -1,35 +1,64 @@
...
+    max-width: fit-content;
   }
-  .issue-list-navbar {
-    order: 0;
+
+  #issue-filters > .issue-list-toolbar-left {
+    margin-top: 10px !important;
+  }
+
+  .ui.stackable.menu:not(.no-vertical-tabs) > .dropdown.item {
```
Comment (wxiaoguang) [inline web_src/css/repo/issue-list.css]: What is `no-vertical-tabs`? I can't find it in code or in Fomantic (correct me if I am wrong)

**Comment by puni9869** [comment]
> Thank you for inviting me for reviewing, and thank you for your contribution 🙏 . I am pretty busy on this one (`25330`) and its related problems, I haven't got enough time for other PRs at the moment.
> 
> Overall I think I am neutral to these changes, while I am not sure whether different languages would also affect the layout. I might take a look later but I can't promise because I am going be in travel in the following weeks.

No worries, whenever required I am available for making changes and discussion to taking this in right direct.
Thanks for acknowledging my pr.

**Review by wxiaoguang — COMMENTED** [review]
Some nits and questions

File: web_src/css/repo/issue-list.css (hunk 8 above)
Comment (puni9869) [reply web_src/css/repo/issue-list.css]: Changed, I took it from chrome[Dev tools styles selector]

File: templates/repo/issue/list.tmpl (hunk 6 above)
Comment (puni9869) [reply templates/repo/issue/list.tmpl]: `stackable ` Removed.

File: templates/repo/issue/list.tmpl (hunk 7 above)
Comment (puni9869) [reply templates/repo/issue/list.tmpl]: Chenged.

File: web_src/css/repo/issue-list.css (hunk 3 above)
Comment (puni9869) [reply web_src/css/repo/issue-list.css:41]: I have changed to shorter and concise query selector.

**Comment by wxiaoguang** [comment]
Two nits (not sure whether they are related): the checkbox, the unnecessary scrollbar.

image

**Comment by puni9869** [comment]
> Two nits (not sure whether they are related): the checkbox, the unnecessary scrollbar.
> 
> image

Fixes
image

image

But  when clicking the checkbox new options are populating.
This also needs a fix. will work on it.
image

**Comment by denyskon** [comment]
This PR seems to break the changes introduced in #25315 by removing used CSS classes....

**Comment by puni9869** [comment]
> This PR seems to break the changes introduced in #25315 by removing used CSS classes....

I can not say right now. But I have gone through that PR. I am making changing cautiously so that they will not effect others.

**Comment by puni9869** [comment]
Desktop:
image

Mobile:- 
image

1024px: 
image


image

image

**Comment by denyskon** [comment]
I think I have a slightly cleaner fix for it, will push it later

**Comment by silverwind** [comment]
Yeah, let's go with https://github.com/go-gitea/gitea/pull/25368. @puni9869 sorry about it. Maybe next time you find some easier issues to work on :)

**Pull request closed without merging**